### ISSUE TITLE
Jasmine tests not running

### ISSUE THREAD

**Issue by Souravirus** [publiclab/plots2#2967]
Report is: 
Running "/home/sourav/.phantomjs/2.1.1/x86_64-linux/bin/phantomjs"  "/home/sourav/.rvm/gems/ruby-2.4.1/gems/jasmine-rails-0.14.7/lib/jasmine_rails/../assets/javascripts/jasmine-runner.js" "file:///home/sourav/Documents/gsoc/plots2/tmp/jasmine/runner.html?root=/home/sourav/Documents/gsoc/plots2&spec="
Running: file:///home/sourav/Documents/gsoc/plots2/tmp/jasmine/runner.html?root=/home/sourav/Documents/gsoc/plots2&spec=
Starting...

Finished
-----------------
0 specs, 0 failures in 0.061s.

ConsoleReporter finished

No specs are running

**Comment by Souravirus** [comment]
@jywarren Have a look at this.

**Comment by Souravirus** [comment]
No, its not related to rails 5. As I have seen the commits before the rails 5.0 upgrade and it was not running even on them.

**Comment by jywarren** [comment]
hmm, we've had trouble keeping these running. Does it run locally? It seems
like phantomjs executes, but doesn't find the tests?

**Comment by Souravirus** [comment]
No not running locally also

**Comment by Souravirus** [comment]
I was using the command as is used in the test `rake spec:javascript`

**Comment by jywarren** [comment]
OK, good to know. Would you mind looking back at when this stopped working so we can pinpoint the commit that caused it?

**Comment by Souravirus** [comment]
Yeah I would search for it. Looks like I have to use binary search for it. :laughing: :laughing:

**Comment by jywarren** [comment]
yeah, sorry, it's a pain!

**Comment by Souravirus** [comment]
no problem

**Comment by Souravirus** [comment]
Here it occured due to this commit:
https://github.com/publiclab/plots2/commit/10ab8331715158f293f20f95cf9e3a5358cebc5e
Sadly, I was my own PR. But I haven't changed the jasmine test as you can see. What might be the problem?

**Comment by jywarren** [comment]
hey, it's not a blame game, just a detective game! 😄 🔍 

Is it somehow possible that the use of `class Reset {` overwrote a global variable that Jasmine uses, also called `Reset`? Could we try changing the name to something else, like `EmailValidator`?

**Comment by Souravirus** [comment]
Yeah ok trying this fix and lets see what happens. Thanks for the help!!

**Comment by Souravirus** [comment]
Nopes it didn't worked

**Comment by Souravirus** [comment]
@jywarren now this is interesting, jasmine is finding the specs when run through web browser. Here is the screenshot:
![screenshot from 2018-07-04 03-01-32](https://user-images.githubusercontent.com/17192149/42245532-c0f40154-7f36-11e8-8027-fc5b039975c3.png)

**Comment by Souravirus** [comment]
I got something https://github.com/searls/jasmine-rails/issues/220 . I guess since in that PR I used modern ES features that's why the phantomjs is unable to find the jasmine tests and it chokes. So, thats why I guess jasmine tests are not running. I think we must move on to another alternative. What do you think?

**Comment by jywarren** [comment]
besides jasmine-rails or beyond Jasmine itself? Sure, what options do we
have?

**Comment by jywarren** [comment]
awesome!

**Comment by Souravirus** [comment]
I am not getting the perfect tests. Tried karma, mocha and teaspoon-jasmine but there are some issues with each of them.

**Comment by Souravirus** [comment]
ok I am currently setting up mocha in rails app.
